### ISSUE TITLE
[Jakarta EE Mailing List Inventory Adjustments] Jakarta WebSocket

### ISSUE THREAD

**Issue by aeiras** [jakartaee/jakarta.ee#993]
https://jakarta.ee/connect/mailing-lists/

11/16.2020 status:
EE4J Eclipse Project for WebSocket developer discussions

**email:** websocket-dev@eclipse.org (43 subscribers)

**Change description to** Discussions related to Server and Client endpoints for WebSocket protocol.

**Comment by chrisguindon** [comment]
Thanks @aeiras,

Do you think you can provide a PR with all the changes that you are [requesting](https://github.com/jakartaee/jakarta.ee/issues/created_by/aeiras)? 
https://github.com/jakartaee/jakarta.ee/blob/src/layouts/shortcodes/pages/mailing-lists.html

I am thinking that we can solve all ~40 issues with 1 PR.

**Comment by aeiras** [comment]
Omgosh, 1 PR would be amazing. ❤️

Yes, but only after there is community review. 
I have used the descriptions as listed in the specifications page but I don’t think most as correct. It needs the community working on the API’s to put some effort. 

Thanks Chris, 🤗

**Comment by aeiras** [comment]
Chris, with a few Jakartees, we already started reviewing the issues and commits are arriving to the PR :) 

I am stating that only you and your team can merge the commits. If there is anything off about my understanding, please do let us know via that ticket. Only those who do are deserving of feedback on actions taken! Thank you @chrisguindon!
https://github.com/jakartaee/jakarta.ee/issues/977#issuecomment-729085045

**Comment by chrisguindon** [comment]
Thanks @aeiras 

I will review and merge these requests as they come in!

The only thing I would add to your comment is that @ivargrimstad is also a committer on the website and can merge PR .

**Comment by cesarhernandezgt** [comment]
Hi @chrisguindon,
 
Does an update on the description for this specification description in https://jakarta.ee/connect/mailing-lists/ page will be automatically propagated to https://accounts.eclipse.org/mailing-list/websocket-dev page ?

![image](https://user-images.githubusercontent.com/1454523/99453677-eb4bbe00-28ea-11eb-98f0-5dee941a2ba2.png)

**Comment by chrisguindon** [comment]
> Does an update on the description for this specification description in https://jakarta.ee/connect/mailing-lists/ page will be automatically propagated to https://accounts.eclipse.org/mailing-list/websocket-dev page ?

No, the content on  https://jakarta.ee/connect/mailing-lists/  is static HTML but the content from https://accounts.eclipse.org/mailing-list/websocket-dev is stored in a database.

However, we do offer an API that we can use to fetch this information which we could use on jakarta.ee 
https://api.eclipse.org/#operation/FetchMailingLists

We can definitely help with updating that information in our database. Would it be possible to share a spreadsheet with all the new text?

Something like this could work:

| list_name     | list_description                                         | list_short_description varchar(255) |
|---------------|----------------------------------------------------------|-------------------------------------|
| websocket-dev | EE4J Eclipse Project for WebSocket developer discussions | websocket developer discussions     |

**Comment by aeiras** [comment]
Hola Chris, @chrisguindon,

using the API would be amazing after the mailing list is updated, just awesome!

Now, do you mean an excel sheet with the 3 columns?  or do you need something else using git?

If excel, I volunteer to populate it as I will be working with whoever chooses to help update stuff via the 41 issues that end with PRs. It seems that 1 PR for all is not possible. As you saw with the other PR, different Jakartees worked on the issue before submitting an updated email via PR. :) So, we have work to do. 

I think Cesar will follow on the technical side!

**Comment by chrisguindon** [comment]
> Now, do you mean an excel sheet with the 3 columns? or do you need something else using git?

Excel would work but Google Sheets might be easier for collaboration. I will need to import the data in our database and I don't plan on scripting this since there are only 41 to update. It will probably quicker for me to simply copy and paste.

We can keep the 41 issues if you need them to push this initiative forward.I am just concern that we might lose track of all the other issues now that they are at the bottom of the issue list.

**Comment by aeiras** [comment]
Point taken, Chris. How about this:

Can you forget completely about the 41 issues? 
Can you only care about PRs submitted? 

If YES, then we clear miscommunications and avoid wasting your time. :) 
We will close the 41 issues as PRs are sent.

 Next week, at the Friends of Jakarta EE, Cesar will demo how to submit PRs and show how to help on this task. Specially, we are thinking about contributors who do not have commit access and are encountering issues. 

Yes, google sheet and Jakarta EE community drive are what I had in mind. I will submit a new issue to provide the notification of the document in a bit. 

Always happy to collaborate with you Chris, thank you!

**Comment by chrisguindon** [comment]
> Point taken, Chris. How about this:
> 
> Can you forget completely about the 41 issues?
> Can you only care about PRs submitted?

Yes, I can definitely work around this. I am raising this for people that might not follow the issue queue as close as me. 

> 
> If YES, then we clear miscommunications and avoid wasting your time. :)
> We will close the 41 issues as PRs are sent.

I don't see this as a waste of time! I think it's good to also have these conversations regardless of the outcome! At the end of the day, WE (collectively), decide how we want to use the issue queue. 

> Next week, at the Friends of Jakarta EE, Cesar will demo how to submit PRs and show how to help on this task. Specially, we are thinking about contributors who do not have commit access and are encountering issues.

That's awesome to hear! 

> Yes, google sheet and Jakarta EE community drive are what I had in mind. I will submit a new issue to provide the notification of the document in a bit.

Sounds good! Feel free to ping me whenever we are ready to update our database!

> Always happy to collaborate with you Chris, thank you!

It's always a pleasure to work with you :)

**Comment by aeiras** [comment]
Feeling deeply appreciative of you Chris!
You are **always** looking into understanding stuff to execute/collaborate better. When you are involved, everything collaborating wise is better. So much value and super fun!  

I will update the Jakarta EE Community via the thread-forum ongoing on this task so that anyone interested in helping learns from this ticket what's what! 

Of course, I will bcc you. You are not expected to even look at that thread. I want you to know when that update is sent so you can be assured that we are paying attention to how best to contribute! :) 

Stay healthy and amazing!

**Comment by aeiras** [comment]
email sent https://www.eclipse.org/lists/jakarta.ee-community/msg02568.html

**Comment by cesarhernandezgt** [comment]
@chrisguindon @aeiras 

I've catchup with the exchanges. As per my understanding:
1. The Github Issues will remain since individual conversations may occur for specific specifications.
2. The "actual" update work has been moved now to fill in the google spreadsheet. This data will then be updated in the API DB.
3. There will no need for individual Pr's since in the near future the page https://jakarta.ee/connect/mailing-lists/  will pull the data from the DB.

If I nailed in all the previous three points, I think the original scope of the upcoming week's demo won't be aligned and can confuse potential contributors in step #2. I'll adjust the demo in another area of the Jakarta EE website so this doesn't collide.

**Comment by aeiras** [comment]
Cesar, 

1- yes, 
2- the google sheet comes only after the PRs are submitted, which means that issues are closed as well. Step 3 before step 4 occurs. 
3- PRs are still needed and only Chris and Ivar can merge those. 
The google sheet will help push the API update across the EF database.

**Comment by autumnfound** [comment]
@aeiras you may have noticed I added a label to all of the issues around the mailing list that were created this past week. This is to help both sides of the coin keep our pages a little cleaner and filter the noise! GitHub has a feature where you can filter issues by labels (among other things) within the main issues window. This lets the developers in the Foundation filter out the community-run initiatives to narrow down on issues in relation to the site. This also works the other way, and allows people interested in community tasks to cut out the noise made while maintaining the site! The default view will always show all, so these will always be around and not lost. This will let both groups work in harmony without having to worry about losing track of things.

**Comment by aeiras** [comment]
Hola Martin, 

I see it and think it is pretty fantastic! Thank you! :)

**Comment by aeiras** [comment]
websocket-dev@eclipse.org	Jakarta Websocket. API for Server and Client endpoints for Websocket protocol.

**Comment by aeiras** [comment]
Martin and Chris, 

I will follow up in the ticket that has the google sheet¡ Thank you for being extra patient. I am closing the tickets while adding the data to the sheet.

**Comment by aeiras** [comment]
Hola @chrisguindon,

There is only 1 ticket left, the google sheet. :)

**Comment by chrisguindon** [comment]
Hi @aeiras 

Does this mean that this is ready for us to update our database or do you mean you are still working on it?

**Comment by aeiras** [comment]
Hola Chris,

I would appreciate your input about tab Jakarta EE [google sheet](https://docs.google.com/spreadsheets/d/1JB-dvwp2Dd9kkiEfF8kHJPlsHRbMe6NZfXo52G0eot8/edit#gid=0). Note that I left blank the List_short_description_varchar (255), Do we need to repeat the same information from the description?

The other tabs are there for later discussion as new projects will need to figure out if the need for a new mailing list is beneficial, etc.

I have recommended that the EE4J stays within the EF infra website as located[ here ](https://projects.eclipse.org/projects/ee4j/contact)  Yet no feedback has been received on that recommendation yet. 

The update message to the WG was sent on Feb 15th https://www.eclipse.org/lists/jakarta.ee-wg/msg00621.html

Thanks for looking into the Jakarta EE tab. I will adjust stuff if necessary to enable the update.
